src/java.base/aix/classes/sun/nio/ch/NioSocketImpl.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) 2019, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package sun.nio.ch;
+
+import java.io.FileDescriptor;
+import java.net.StandardSocketOptions;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.SocketException;
+import java.lang.ref.Cleaner.Cleanable;
+import jdk.internal.access.JavaIOFileDescriptorAccess;
+import jdk.internal.access.SharedSecrets;
+import java.net.SocketImpl;
+import sun.net.PlatformSocketImpl;
+import java.net.SocketAddress;
+import java.net.InetAddress;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.SocketOption;
+import java.util.Objects;
+import java.util.Set;
+import java.net.InetSocketAddress;
+
+public final class NioSocketImpl extends SocketImpl implements PlatformSocketImpl {
+
+    private static final NativeDispatcher nd = new SocketDispatcher();
+    private final FileDescriptor fd = new FileDescriptor();
+
+   // private final FileDescriptor fd;
```

Comment:
i see few comments with funny alignment, maybe fix them as well before making it rfr :-)